core/src/main/java/hudson/model/Queue.java

```diff
@@ -1535,32 +1573,27 @@ public void maintain() {
             ConsistentHash<Node> hash = new ConsistentHash<Node>(NODE_HASH);
             hash.addAll(hashSource);
 
-            Label lbl = p.getAssignedLabel();
             for (Node n : hash.list(p.task.getFullDisplayName())) {
                 final Computer c = n.toComputer();
-                if (c==null || c.isOffline())    continue;
-                if (lbl!=null && !lbl.contains(n))  continue;
-                if (n.canTake(p) != null) continue;
+                if (n.canTake(p) != null) {
+                    continue;
+                }
+                if (c == null || c.isOffline()) {
```

Comment:
Computer should be checked before the node. If the computer has been already deleted, there is no need to check the node at all